### PR TITLE
No more image rotation in setImage

### PR DESCRIPTION
**Pull request by ws233**
1. No more redrawing an image to the orientationUp, if its in different orientation. Instead make a Pix directly from rotated image.
2. UIImage + G8FixOrientation has been removed cause it's unnecessary now.
3. Added tests for all available image orientations for 8 and 32 bpp.

### CODE REVIEW

File: TesseractOCR/G8Tesseract.mm [hunk 1]
```diff
@@ -819,14 +816,19 @@ - (Pix *)pixForImage:(UIImage *)image
     const UInt8 *pixels = CFDataGetBytePtr(imageData);
 
     size_t bitsPerPixel = CGImageGetBitsPerPixel(cgImage);
+    int bytesPerPixel = (int)bitsPerPixel / 8;
```
Comment (kevincon) [inline TesseractOCR/G8Tesseract.mm]: This can't be negative and since it's more of a size anyway, I think it should be a `size_t` as well.

**Comment by kevincon** [comment]
This is fantastic, really great work here. :) 

Can you take another look at the datatypes you're using for these calculations, though? I see a lot of `int`'s used with respect to `copyBlock` when it looks like it would never be called with negative values. As much as possible I think we should use unsigned types if the data really will always be unsigned.

Also the `toAddr` parameter being `l_uint32` and the `fromAddr` parameter being `UInt8` seems really weird to me; can't they have the same type?

**Comment by ws233** [comment]
I've fixed types. Pls, review.

@kevincon, actually, we may have the same types of pointers in the copyBlock, but it causes a lot of type casting inside blocks. So I prefer to define different pointer types. We need this because we consider the "from" memory as an array of chars, since we need to swap bytes to a correct endianness. At the same time we consider "to" memory as an integer, since we need to set it at once, after we swapped bytes in correct order. Let me know, if you think different.

**Comment by kevincon** [comment]
Looks good! Merging and closing.

**Pull request merged**